engine/render/src/render/render_script_camera.cpp

```diff
@@ -0,0 +1,314 @@
+    *
+    * `fov`
+    * : [type:number] the field of view.
+    *
+    * `aspect_ratio`
+    * : [type:number] the aspect ratio.
+    *
+    * `near_z`
+    * : [type:number] the near z.
+    *
+    * `far_z`
+    * : [type:number] the far z.
+    *
+    * `orthographic_projection`
+    * : [type:boolean] true if the camera is using an orthographic projection.
+    *
+    * `auto_aspect_ratio`
+    * : [type:boolean] true if the camera is using an automatic aspect ratio.
+    *
+    * @name camera.get_cameras
+    * @return cameras [type:table] a table with all camera URLs
+    *
+    * @examples
+    * ```lua
+    * local info = camera.get_info("main:/my_go#camera")
+    * render.set_camera(info.handle)
+    * ```
+    */
+    static int RenderScriptCamera_GetInfo(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+        RenderCamera* camera = CheckRenderCamera(L, 1, g_RenderScriptCameraModule.m_RenderContext);
+
+        lua_newtable(L);
+        lua_pushstring(L, "url");
+        dmScript::PushURL(L, camera->m_URL);
+        lua_settable(L, -3);
+
+        lua_pushstring(L, "projection");
+        dmScript::PushMatrix4(L, camera->m_Projection);
+        lua_settable(L, -3);
+
+        lua_pushstring(L, "view");
+        dmScript::PushMatrix4(L, camera->m_View);
+        lua_settable(L, -3);
+
+        lua_pushstring(L, "handle");
+        lua_pushnumber(L, camera->m_Handle);
+        lua_settable(L, -3);
+
+        // TODO: Since we don't have any way of changing this (yet), we don't expose it.
+        //       The idea is to use a normalized viewport vector that can be set though the API
+        //       and via the editor, but it is the next part of this feature.
+        /*
+        lua_pushstring(L, "viewport");
+        dmScript::PushVector4(L, camera->m_Data.m_Viewport);
+        lua_settable(L, -3);
+        */
+
+        lua_pushstring(L, "main_camera");
+        lua_pushboolean(L, camera->m_IsMainCamera);
+        lua_settable(L, -3);
+
+    #define PUSH_NUMBER(name, param) \
+        lua_pushstring(L, name); \
+        lua_pushnumber(L, camera->m_Data.param); \
+        lua_settable(L, -3);
+
+        PUSH_NUMBER("fov",          m_Fov);
+        PUSH_NUMBER("aspect_ratio", m_AspectRatio);
+        PUSH_NUMBER("near_z",       m_NearZ);
+        PUSH_NUMBER("far_z",        m_FarZ);
+    #undef PUSH_NUMBER
+
+    #define PUSH_BOOL(name, param) \
+        lua_pushstring(L, name); \
+        lua_pushboolean(L, camera->m_Data.param); \
+        lua_settable(L, -3);
+
+        PUSH_BOOL("orthographic_projection", m_OrthographicProjection);
+        PUSH_BOOL("auto_aspect_ratio",       m_AutoAspectRatio);
+    #undef PUSH_BOOL
+
+        return 1;
+    }
+
+    // This is called when getting a property on the camera module.
+    // If the 'MAIN' property is called, we return the URL to the current main camera.
+    // Otherwise, we just return the tble entry.
+    static int RenderScriptCamera_index(lua_State* L)
```

Comment:
Tbh I don't like this `camera.MAIN` property at all. I would prefer to have it like a separate function to make it clear and in alight with the rest engine's API

Reply:
Yes, this is the thing that stands out the most with this solution. It feels like something quite different from what we have elsewhere.

Reply:
Hmm, I think perhaps we should remove the concept of having a main camera altogether. The point of having a "main" camera is to organize your rendering in a multi-camera setting, where you want one camera to be a "shadow" camera and some other camera to be a "scene" camera, and another to be a "postprocessing" camera or something like that. I think instead of a specific "main" camera like this, I think you would want to tag a camera with some made-up tag, such as "scene", "shadow", "postprocessing".. For now lets skip the main camera setup